src/Microsoft.AspNet.Hosting/Startup/StartupLoader.cs

```diff
@@ -94,9 +90,9 @@ private object Invoke(MethodInfo methodInfo, object instance, IApplicationBuilde
             string environmentName,
             IList<string> diagnosticMessages)
         {
-            if (String.IsNullOrEmpty(applicationName))
+            if (string.IsNullOrEmpty(applicationName))
             {
-                return _next.LoadStartup(applicationName, environmentName, diagnosticMessages);
+                throw new ArgumentNullException("applicationName");
```

Comment:
Wrong exception type. This is thrown for null or empty, yet the exception is for null only. Use a regular ArgException with a message that says can't be null or empty (copy whatever message we used in WSR).